x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SearchableSnapshotAction.java

```diff
@@ -206,12 +205,17 @@ public String getSnapshotRepository() {
             snapshotRepository);
         CleanupSnapshotStep cleanupSnapshotStep = new CleanupSnapshotStep(cleanSnapshotKey, createSnapshotKey, client);
         AsyncActionBranchingStep createSnapshotBranchingStep = new AsyncActionBranchingStep(
-            new CreateSnapshotStep(createSnapshotKey, mountSnapshotKey, client), cleanSnapshotKey, client);
+            new CreateSnapshotStep(createSnapshotKey, waitForDataTierKey, client), cleanSnapshotKey, client);
 
-        // Now mount the snapshot to create the new index, if the skipGeneratingSnapshotStep determined a snapshot already existed that
+        MountSearchableSnapshotRequest.Storage storageType = getConcreteStorageType(mountSnapshotKey);
+
+        // If the skipGeneratingSnapshotStep determined a snapshot already existed that
         // can be used, it jumps directly here, skipping the snapshot generation steps above.
+        WaitForDataTierStep waitForDataTierStep =
+            new WaitForDataTierStep(waitForDataTierKey, mountSnapshotKey,
+                MountSearchableSnapshotRequest.getDataTiersPreference(storageType));
```

Comment:
This is technically going to be sending the wrong preference, because the preference is either `data_cold,data_warm,data_hot` or `data_frozen` from the `getDataTiersPreference`, but actually if this is the hot phase, it will end up with a different preference:

https://github.com/elastic/elasticsearch/blob/c5b9fdea5bf7ac5988f7dde71a61754a1a3b50fb/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/MountSnapshotStep.java#L108-L111

Reply:
thanks for noticing, fixed in 3d23644c27086a84dcb88c79a99fc6644da675f8